ethereum/core/src/main/java/org/hyperledger/besu/ethereum/mainnet/DepositsValidator.java

```diff
@@ -64,10 +77,41 @@ public boolean validateDepositsRoot(final Block block) {
   class AllowedDeposits implements DepositsValidator {
 
     private static final Logger LOG = LoggerFactory.getLogger(AllowedDeposits.class);
+    private final Address depositContractAddress;
+
+    public AllowedDeposits(final Address depositContractAddress) {
+      this.depositContractAddress = depositContractAddress;
+    }
+
+    public Address getDepositContractAddress() {
+      return depositContractAddress;
+    }
 
     @Override
-    public boolean validateDeposits(final Optional<List<Deposit>> deposits) {
-      return true; // TODO 6110: Validate deposits in the next PR
+    public boolean validateDepositParameter(final Optional<List<Deposit>> deposits) {
+      return deposits.isPresent();
+    }
+
+    @Override
+    public boolean validateDeposits(final Block block, final List<TransactionReceipt> receipts) {
+      if (block.getBody().getDeposits().isEmpty()) {
+        LOG.warn("Deposits must not be null when Deposits are activated");
+        return false;
+      }
+
+      List<Deposit> actualDeposits = new ArrayList<>(block.getBody().getDeposits().get());
+      List<Deposit> expectedDeposits = new ArrayList<>();
+
+      for (TransactionReceipt receipt : receipts) {
+        for (Log log : receipt.getLogsList()) {
+          if (depositContractAddress.equals(log.getLogger())) {
+            Deposit deposit = DepositDecoder.decodeFromLog(log);
+            expectedDeposits.add(deposit);
+          }
+        }
+      }
+
+      return actualDeposits.equals(expectedDeposits);
     }
 
     @Override
```

Comment:
Not part of this PR but part of the DepositsValidator code. In validateDepositsRoot the error message when the deposits root doesn't match has "Invalid block: transaction root mismatch" instead of "deposit root mismatch"